tests/pmap_test.py

```diff
@@ -352,6 +352,25 @@ def sum_and_broadcast(x, axis):
     expected = sum_and_broadcast(sum_and_broadcast(x, 0), 1)
     self.assertAllClose(ans, expected, check_dtypes=False)
 
+  def testPsumReplicaGroups(self):
+    replicas = xla_bridge.device_count()
+    if replicas % 2 != 0:
+      raise SkipTest
+    replica_groups = onp.arange(replicas).reshape(2, replicas // 2).tolist()
+    f = pmap(lambda x: x - lax.psum(x, 'i', replica_groups), axis_name='i')
```

Comment:
Do replica_groups work with nested pmaps?

Reply:
I'd say no, not really (I mean, it'll work but it'll use the computation-global replica IDs, breaking the nested pmap abstraction). I guess we can either support per-axis replica groups (not sure how hard that would be, but I don't want to underestimate things that touch nested pmap) or we could throw an error if you use replica_groups in a nested pmap?